### ISSUE TITLE
[Rust] Implement new Concept Exercise: numbers

### ISSUE THREAD

**Issue by lewisclement** [exercism/v3#750]
This issue describes how to implement the `numbers` concept exercise for the <LANG> track.

## Goal

The goal of this exercise is to teach how numbers are implemented in Rust. Because Rust doesn't have implicit casting, this automatically includes teaching about primitive bit sizes. 

## Things to teach

- Know how number literals are represented in Rust
- Know the different types of integer and floating point primitives
- Know how to cast

## Things not to teach
- Arithmetic
- Definitely not arithmetic traits like `Add` and `Sub`, best to introduce with methods
- Mutable variables

## Concepts

- Numbers
- Integers
- Floating point values (basic)

## Prerequisites

None

### Hints

[Rust book - Scalar types](https://doc.rust-lang.org/stable/book/ch03-02-data-types.html?highlight=primitive#scalar-types)
[cheats.rs - Basic Types](https://cheats.rs/#basic-types)
[Rust reference - Integer literals](https://doc.rust-lang.org/stable/reference/tokens.html#integer-literals)
[Rust reference - Floating point literals](https://doc.rust-lang.org/stable/reference/tokens.html#floating-point-literals)

### After

[Rust reference - Literals](https://doc.rust-lang.org/stable/reference/expressions/literal-expr.html)
[Rust reference - Numeric types](https://doc.rust-lang.org/stable/reference/types/numeric.html)
[Rust reference - Type cast expressions](https://doc.rust-lang.org/stable/reference/expressions/operator-expr.html#type-cast-expressions)

## Representer

This exercise does not require any specific representation logic to be added to the representer.

## Analyzer

This exercise does not require any specific logic to be added to the analyzer.

## Implementing

To implement a Clojure concept exercise, the following files must be created:
```
languages
└── rust
    └── exercises
        └── concept
            └── <SLUG>
                ├── .docs
                |   ├── instructions.md
                |   ├── introduction.md
                |   ├── hints.md
                |   └── after.md (optional)
                ├── .meta
                |   |── design.md
                |   |── config.json
                |   └── example.rs
                ├── src
                |   └── lib.rs
                ├── cargo.toml
                └── tests
                    └── <NAME>.rs
```

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by workingjubilee** [comment]
To implement a Clojure concept exercise, huh. :^)

I think I can take a crack at this.

**Comment by workingjubilee** [comment]
This should have the track/rust label so it's more findable, especially in case I fall on my sword and don't finish.

**Comment by workingjubilee** [comment]
Oh, silly me, this should be closed, since #825 resolves it.

**Comment by lewisclement** [comment]
Yes, it should! Sorry for the bother! There should be new issues up soon!
